generatebundlefile/promote.go

```diff
@@ -0,0 +1,152 @@
+		}
+	}
+	return clients, nil
+}
+
+// PromoteHelmChart will take a given repository, and authFile and handle helm and image promotion for the mentioned chart.
+func (c *SDKClients) PromoteHelmChart(repository, authFile string, profile bool) error {
+	var name, version, sha string
+	pwd, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("Error getting pwd %s", err)
+	}
+
+	//
+	if profile {
+		name, version, sha, err = c.getNameAndVersionPublic(repository, c.stsClient.AccountID)
+		if err != nil {
+			return fmt.Errorf("Error getting name and version from helmchart %s", err)
+		}
+	} else {
+		name, version, sha, err = c.getNameAndVersion(repository, c.stsClient.AccountID)
+		if err != nil {
+			return fmt.Errorf("Error getting name and version from helmchart %s", err)
+		}
+	}
+
+	// Pull the Helm chart to Helm Cache
+	BundleLog.Info("Promoting chart and image version", name, version)
+	semVer := strings.Replace(version, "_", "+", 1) // TODO use the Semvar library instead of this hack.
+	chartPath, err := PullHelmChart(name, semVer, authFile)
+	if err != nil {
+		return fmt.Errorf("Error pulling helmchart %s", err)
+	}
+	// Get the correct Repo Name from the flag based on ECR repo name formatting
+	// since we prepend the github org on some repo's, and not on others.
+	chartName, helmname, err := splitECRName(name)
+	if err != nil {
+		return fmt.Errorf("Failed splitECRName %s", err)
+	}
+	// Untar the helm .tgz to pwd and name the folder to the helm chart Name
+	dest := filepath.Join(pwd, chartName)
+	err = UnTarHelmChart(chartPath, chartName, dest)
+	if err != nil {
+		return fmt.Errorf("failed pulling helm release %s", err)
+	}
+
+	// Check for requires.yaml in the unpacked helm chart
+	helmDest := filepath.Join(pwd, chartName, helmname)
+	f, err := hasRequires(helmDest)
+	if err != nil {
+		return fmt.Errorf("Helm chart doesn't have requires.yaml inside %s", err)
+	}
+
+	// Unpack requires.yaml into a GO struct
+	helmRequires, err := validateHelmRequires(f)
+	if err != nil {
+		return fmt.Errorf("Unable to parse requires.yaml file to Go Struct %s", err)
+	}
+	// Add the helm chart to the struct before looping through lookup/promote since we need it promoted too.
+	helmRequires.Spec.Images = append(helmRequires.Spec.Images, Image{Repository: chartName, Tag: version, Digest: sha})
+
+	// Change the source destination check depending on release or not
+	destination := c.ecrPublicClient
+	if profile {
+		destination = c.ecrPublicClientRelease
+	}
+
+	// Loop through each image, and the helm chart itself and check for existance in ECR Public, skip if we find the SHA already exists in destination.
+	// If we don't find the SHA in public, we lookup the tag from Private, and copy from private to Public with the same tag.
+	for _, images := range helmRequires.Spec.Images {
+		check, err := destination.shaExistsInRepository(images.Repository, images.Digest)
```

Comment:
Is it really ok if all of these fail? You log some things, but you don't error out. The user could potentially miss the logs and assume that everything was ok.

Reply:
Added return errors for these in commit.